### PR TITLE
CI Windows fixes

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

A couple of tests are failing for the 4.2 and devel Windows tests in CI (GHA) with some recent changes in the Rtools42 toolchain. Attempting to address them.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)    
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by jgabry** [comment]
Thanks!

File: R/csv.R [hunk 1]
```diff
@@ -227,8 +227,19 @@ read_cmdstan_csv <- function(files,
...
+      grep_path_repaired <- withr::with_path(
+        c(
+          toolchain_PATH_env_var()
+        ),
+        repair_path(Sys.which("grep.exe"))
+      )
+      grep_path_quotes <- paste0('"', grep_path_repaired, '"')
+      fread_cmd <- paste0(
+        grep_path_quotes,
+        " '^[#a-zA-Z]' --color=never '",
```
[the remark this answers is not in the document]
Comment (weshinsley) [reply R/csv.R:239]: Without that, I get an error of the form:
`grep.exe: 'T:/Wes/file.csv': Invalid argument`

File: R/csv.R [hunk 2]
```diff
@@ -558,8 +569,19 @@ read_csv_metadata <- function(csv_file) {
...
+      ),
+      repair_path(Sys.which("grep.exe"))
+    )
+    grep_path_quotes <- paste0('"', grep_path_repaired, '"')
+    fread_cmd <- paste0(
+      grep_path_quotes,
+      " '^[#a-zA-Z]' --color=never '",
+      csv_file,
+      "'"
+    )
```
Comment (weshinsley) [inline R/csv.R:584]: Similar to above - my windows setup needs double-quoting on 581 and 583

File: R/csv.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (rok-cesnovar) [reply R/csv.R:239]: Thanks!

[the remark this answers is not in the document]
Comment (weshinsley) [reply R/csv.R:239]: Not relevant to this PR, but additionally I found the "classic windows grep" had an issue here
(http://gnuwin32.sourceforge.net/packages/grep.htm v2.5.4 - old, but sometimes included as an attempt to provide all the GNU tools)

`C:/Windows/grep searchfor T:/wes/file.txt` failed, whereas
`grep searchfor T:/wes/file.txt` worked - same grep.exe being used as it was in my path, but the specifying of the drive letter which Sys.which() provided seemed offensive to grep.

I fixed this, unconventionally, by upgrading grep to a binary from 
`https://github.com/mbuilov/grep-windows`
which did not have that problem.

Not quite sure where to note that, but hope it's helpful

[the remark this answers is not in the document]
Comment (weshinsley) [reply R/csv.R:239]: I did wonder whether this portion of code would be better done with some base R, instead of shelling out to grep.exe at all.

**Comment by rok-cesnovar** [comment]
Finally addressed all Windows CI issues. All tests pass, merging now.

**Pull request merged**